standards/application/messaging-api.md

```diff
@@ -0,0 +1,965 @@
+If set to `true` but `confirmContentTopics` are not provided, the received property on the `MessageRecord` (as defined in the `Message Storage API`) MUST be populated only after the `Subscribe API` is called.
+If set to `false`, the received property MUST NOT be populated, unless `Subscribe API` is called.
+
+##### `confirmContentTopics`
+An optional property that SHOULD be provided in conjunction with either `confirmStored` or `confirmReceived`.
+It includes an array of `contentTopics` that the node MUST start monitoring in the background upon initialization,
+as defined in the [TOPICS](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/informational/23/topics.md#content-topic-format) specification.
+
+#### Programmatic API / REST API
+
+No methods or `REST` endpoints are provided for initial configuration.
+Instead, the initial configuration MUST be supplied at node creation time according to the guidelines of the chosen implementation.
+
+### Send
+
+The `Send API` is responsible for broadcasting messages across the network using the configured protocol.
+The node SHOULD select the appropriate protocol based on its configuration and the protocols that are mounted:
+- If the initial configuration specifies mode: "edge", then [LIGHTPUSH](../standards/core/lightpush.md) MUST be used to send messages.
+- If the initial configuration specifies mode: "service", then [RELAY](https://github.com/vacp2p/rfc-index/blob/0277fd0c4dbd907dfb2f0c28b6cde94a335e1fae/waku/standards/core/11/relay.md) MUST be used to send messages.
+
+The `Send API` is also responsible for ensuring message delivery and MUST attempt retries as described in the `Retry` section.
+
+#### Retry
+
+If a message fails to satisfy any of the following conditions, it MUST be automatically resent:
+- The message is not successfully sent using [LIGHTPUSH](../standards/core/lightpush.md) or [RELAY](https://github.com/vacp2p/rfc-index/blob/0277fd0c4dbd907dfb2f0c28b6cde94a335e1fae/waku/standards/core/11/relay.md).
+- The message is not acknowledged by `Message Storage API` described below.
+- Retry attempts MUST NOT continue if the node’s health state changes to unhealthy (as defined in the `Health API` described below).
+- Retry attempts MUST resume when the node’s health state transitions to either `minimally healthy` or `healthy` defined in the `Health API` below.
+
+Retry attempts MUST occur in the background using a fixed retry count and exponential backoff, as follows:
+- A total of 3 retry attempts MUST be made following the initial send attempt.
+- The retry intervals MUST be 1 second, then 2 seconds, and finally 4 seconds, respectively.
+- The decision to retry MUST be based on the underlying protocol’s response from a remote peer.
```

Comment:
Two comments here:

1. This can be refer back to the p2p reliability spec
> The usual remedial action is to retransmit the message. 

I would not re-state this logic here. But instead, just clarifying that `send` API is expected to implement the strategies configured and stated in p2p reliability state.

2. Actual back-off timing can be left to implementater's choice

Reply:
If there is one thing you should include or that cna be configurable is for how long do we retry - because this may be application dependent. but for now, I'll leave it to the implementation

Reply:
also agree, addressed here https://github.com/waku-org/specs/pull/19/commits/1e4227ac80a507f09f3f3d2c757277c2206a4e6c